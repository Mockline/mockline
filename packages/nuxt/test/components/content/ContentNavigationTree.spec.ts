import { describe, it, expect } from 'vitest'
import type {TypeOf} from "zod";
// @ts-ignore
import { MContentNavigationTree } from '#components'
import ComponentRender from "../../component-render";

describe('ContentNavigationTree', () => {
  it.each([
    ['basic case', { props: { links: [{ title: 'title', path: '/path' }] } }],
    ['<MContentNavigationTree :links="[{ title: \'title\', path: \'/path\' }]">ContentNavigationTree slot</MContentNavigationTree>'],
    // @ts-expect-error
  ])('renders %s correctly', async (nameOrHtml, options: TypeOf<typeof MContentNavigationTree.props>) => {
    const html = await ComponentRender(nameOrHtml, options, MContentNavigationTree)
    expect(html).toMatchSnapshot()
  })
})


