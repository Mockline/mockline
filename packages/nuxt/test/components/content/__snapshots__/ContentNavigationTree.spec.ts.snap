// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContentNavigationTree > renders <MContentNavigationTree :links="[{ title: 'title', path: '/path' }]">ContentNavigationTree slot</MContentNavigationTree> correctly 1`] = `
"<nav class="flex flex-col gap-4">
  <div>
    <h3 class="truncate font-semibold mb-2 text-sm">title</h3>
    <ul class="flex flex-col gap-0.5"></ul>
  </div>
</nav>"
`;

exports[`ContentNavigationTree > renders basic case correctly 1`] = `
"<nav class="flex flex-col gap-4">
  <div>
    <h3 class="truncate font-semibold mb-2 text-sm">title</h3>
    <ul class="flex flex-col gap-0.5"></ul>
  </div>
</nav>"
`;
